filament/backend/CMakeLists.txt

```diff
@@ -98,14 +98,26 @@ if (FILAMENT_SUPPORTS_OPENGL AND NOT FILAMENT_USE_EXTERNAL_GLES3 AND NOT FILAMEN
     elseif (WEBGL)
         list(APPEND SRCS src/opengl/platforms/PlatformWebGL.cpp)
     elseif (LINUX)
-        list(APPEND SRCS src/opengl/platforms/PlatformGLX.cpp)
+        if (FILAMENT_SUPPORTS_WAYLAND)
+            list(APPEND SRCS src/opengl/platforms/PlatformEGLWayland.cpp)
+        elseif (FILAMENT_SUPPORTS_X11)
+            list(APPEND SRCS src/opengl/platforms/PlatformGLX.cpp)
+        endif()
     elseif (WIN32)
         list(APPEND SRCS src/opengl/platforms/PlatformWGL.cpp)
     else()
         list(APPEND SRCS src/opengl/platforms/PlatformDummyGL.cpp)
     endif()
 endif()
 
+if (FILAMENT_SUPPORTS_WAYLAND)
+    add_compile_definitions(
```

Comment:
Can we use `target_compile_definitions` here instead?

Reply:
good point.  I made them target specific and moved them after the backend target def.